packages/plugin-ext/src/plugin/documents.ts

```diff
@@ -81,34 +81,70 @@ export class DocumentsExtImpl implements DocumentsExt {
             this._onDidSaveTextDocument.fire(data.document);
         }
     }
-    $acceptModelWillSave(strUrl: UriComponents, reason: theia.TextDocumentSaveReason): Promise<SingleEditOperation[]> {
-        return new Promise<SingleEditOperation[]>((resolve, reject) => {
-            const uri = URI.revive(strUrl);
-            const uriString = uri.toString();
-            const data = this.editorsAndDocuments.getDocument(uriString);
-            if (data) {
-                const onWillSaveEvent: theia.TextDocumentWillSaveEvent = {
-                    document: data.document,
-                    reason: reason,
-                    /* tslint:disable:no-any */
-                    waitUntil: async (editsPromise: PromiseLike<theia.TextEdit[] | any>) => {
-                        const editsObjs = await editsPromise;
-                        if (this.isTextEditArray(editsObjs)) {
-                            const editOperations: SingleEditOperation[] = (editsObjs as theia.TextEdit[]).map(textEdit => Converter.fromTextEdit(textEdit));
-                            resolve(editOperations);
-                        } else {
-                            resolve([]);
-                        }
+
+    async $acceptModelWillSave(strUrl: UriComponents, reason: theia.TextDocumentSaveReason): Promise<SingleEditOperation[]> {
+        const uri = URI.revive(strUrl).toString();
+        const operations: SingleEditOperation[] = [];
+        let didTimeout = false;
+        const didTimeoutHandle = setTimeout(() => didTimeout = true, 1500);
```

Comment:
Above, you reject with `1750`. Why the difference? Can we use a shared constant instead?

Reply:
there are 2 different processes, server process timeouts in 1750ms, plugin host process tries to timeout a bit earlier to squeeze at least some edits, if it timeouts later no edits will be provided

i could add the constant in some main-ext interfaces, they are shared between these processes.

Reply:
It's fine. I just wanted to understand why the difference between the timeouts. Thanks!